### ISSUE TITLE
Bug(react/primitives): Button isLoading loading text is not announced by screen readers

### ISSUE THREAD

**Issue by hbuchel** [aws-amplify/amplify-ui#4343]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Primitive components

### How is your app built?

Next.js

### What browsers are you seeing the problem on?

Chrome

### Please describe your bug.

Button `loadingText` is not announced by some screen readers when `isLoading` boolean prop for Button is set to true. `loadingText` is not part of an aria-live region (like `aria-live="polite"` or `role="status"` so the change in context is not reliably announced by assistive tech.

Current considerations based on some discussions around refactoring:
- https://github.com/aws-amplify/amplify-ui/pull/3816 
- Refactor loading functionality to allow customers to manage their own live regions and have better control over Loading indicator
- With the previous option, create a LoadingButton for customers who want a default/simple implementation

### What's the expected behaviour?

`loadingText` should be announced when isLoading is true.

### Help us reproduce the bug!

https://codesandbox.io/p/sandbox/button-isloading-9tktqj?file=%2Fsrc%2FApp.tsx%3A37%2C18

In this example, the first Begin button does not announce its loadingText when activated when using NVDA (tested on Chrome/Windows)

In the second example, the loading text is announced.

### Code Snippet

```typescript
// Put your code below this line.

```
